### PR TITLE
fix: add documentation on substitution and shell variable expansion

### PR DESCRIPTION
**Pull request by lucas-deangelis**
As discussed in #37. I've kept the explanation in `.help()` short to avoid either weird formatting or having to switch it to `.long_help()`.

Using double quotes for the REPLACEMENT argument with capture group substitution in shells like Bash or zsh will not work as intended. For example, a "${1}" will be replaced by the content of the shell variable 1 before rnr can read them, resulting in no substitution.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -478,6 +478,13 @@ rnr -f '(\w+)-(\d+).(\w+)' '${2}-${1}.${3}' ./*
 ├── 02-file.txt
 └── 03-file.txt
 ```
+__SHELL NOTE:__ In shells like Bash and zsh, make sure to wrap the `REPLACEMENT` pattern in single quotes. Otherwise, capture group indices will be replaced by expanded shell variables which will most likely be empty:
+```
+# Command typed in the shell
+rnr -f '(\w+)-(\d+).(\w+)' "${2}-${1}.${3}" ./*
+# After parameter expansion, the command that is actually ran:
+rnr -f '(\w+)-(\d+).(\w+)' "-." ./*
+```
```
Comment (ismaelgv) [inline README.md:487]:
```suggestion
__SHELL NOTE:__ In shells like Bash and zsh, make sure to wrap the `REPLACEMENT` pattern in single quotes. Otherwise, capture group indices will be replaced by expanded shell variables.
```
I think this note is fine, but I would remove the example since it may cause confusion. Other examples in the documentation are valid behavior.

File: src/app.rs [hunk 2]
```diff
@@ -88,7 +88,7 @@ pub fn create_app<'a>() -> App<'a, 'a> {
         )
         .arg(
             Arg::with_name("REPLACEMENT")
-                .help("Expression replacement")
+                .help("Expression replacement. Enclose in single quotes to prevent variable expansion")
```
Comment (ismaelgv) [inline src/app.rs:91]: Not sure about this, the use may actually want to use variable expansion, for example:

```sh
rnr -r '.txt' "${OTHER_EXT}" .
```

Maybe we can reword this to:
`Expression replacement (use single quotes for capture groups)`

**Review by ismaelgv — CHANGES_REQUESTED** [review]
Thanks for the contribution :wink:

**Comment by lucas-deangelis** [comment]
Thanks for the review! I've made the two changes.

File: README.md
Comment (ismaelgv) [inline README.md:1]: It seems that some changes were applied here, but this is not the global ToC of the document, just the examples ToC. I would revert this.

Comment (lucas-deangelis) [reply README.md:1]: You're right, good catch and sorry about that. This should be fixed now.

**Review by ismaelgv — APPROVED** [review]
